main.rb

```diff
@@ -0,0 +1,149 @@
+end
+
+def env_has_key(key)
+  value = get_env(key)
+  return value unless value.nil? || value.empty?
+
+  abort("Input #{key} is missing.")
+end
+
+def run_command(command)
+  puts "@@[command] #{command}"
+  unless system(command)
+    exit $?.exitstatus
+  end
+end
+
+def abort_with1(message)
+  puts "@@[error] #{message}"
+  exit 1
+end
+
+def check_config_file(config_path)
+  unless File.exist?(config_path)
+    abort_with1("Sauce config file not found at #{config_path}")
+  end
+end
+
+def auth_saucelabs(username, access_key)
+  run_command('sudo npm install -g saucectl')
+  puts "Saucectl version: #{run_command('saucectl -v')}"
+  run_command("saucectl configure --username #{username} --accessKey #{access_key}")
+end
+
+def run_saucectl(params)
+  command = "cd #{$repo_dir} && sudo saucectl run #{params.join(' ')}"
+  puts "Running command: #{command}"
+  run_command(command)
+end
+
+def check_suite_name(config_path, suite_name)
+  config = YAML.load_file(config_path)
+
+  unless config['suites'].find { |suite| suite['name'] == suite_name }
+    available_suites = config['suites'].map { |suite| suite['name'] }
+    abort_with1("Suite name '#{suite_name}' not found. Available suites: #{available_suites.join(', ')}")
+  end
+end
+
+def update_app_path(config_path)
+  config = YAML.load_file(config_path)
+  new_app_path = get_env("AC_SL_APP_PATH")
+  new_test_app_path = get_env("AC_TEST_APP_PATH")
+
+  if config['espresso']
+    target_config = config['espresso']
+  elsif config['xcuitest']
+    target_config = config['xcuitest']
+  else
+    puts "No recognizable configuration (espresso or xcuitest) found in the YAML file."
+    exit 0
+  end
+
+  if !new_app_path || !new_test_app_path
+    puts "Missing app or testApp path. Please make sure you have built the test app before this step."
+    exit 0
+  end
+
+  target_config['app'] = new_app_path
+  target_config['testApp'] = new_test_app_path
+
+  File.open(config_path, 'w') do |file|
+    file.write(config.to_yaml)
+  end
+
+  puts "Updated .sauce/config.yml with new app and testApp paths."
+end
+
+def add_parameter(command, flag, value)
+  command << "#{flag}=#{value}" if value
+end
+
+def add_bool_parameter(command, flag, value)
+  command << "#{flag}" if value=="true"
+end
+
+def export_sauce_artifacts(download_dir)
+  if download_dir
+      puts "Exporting downloaded sauce files..."
+      run_command("cd $AC_OUTPUT_DIR && mkdir -p sauce && cp -r #{download_dir} sauce")
+  else
+    puts "Download artifacts direction not found for Sauce..."
+  end
+end
+
+$repo_dir = env_has_key("AC_REPOSITORY_DIR")
+sl_rel_config_path = env_has_key("AC_SL_CONFIG_PATH")
+sl_config_path = sl_rel_config_path.nil? ? nil : File.join($repo_dir, sl_rel_config_path)
+check_config_file(sl_config_path)
+sl_username = env_has_key("AC_SL_USERNAME")
+sl_access_key = env_has_key("AC_SL_ACCESS_KEY")
+auth_saucelabs(sl_username, sl_access_key)
+sl_run_only_via_config = get_env("AC_RUN_ONLY_VIA_CONFIG") || "false"
```

Comment:
Since this is a Sauce Labs step-specific variable, it should be named `AC_SL_RUN_ONLY_VIA_CONFIG` similar to the other variables. (_convention tracking_)